core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java

```diff
@@ -59,10 +66,17 @@ public String sql() {
             + projectionSqlString()
             + " FROM "
             + rdbEngine.encloseFullTableName(schema, table);
-    if (isConditionalQuery) {
-      sql += " WHERE " + conditionSqlString();
+    if (isRelationalQuery) {
+      // for relational abstraction
+      sql += relationalConditionSqlString();
```

Comment:
How about using StringBuilder instead to avoid instantiating temporary objects? Using StringBuilder is basically better in terms of performance and memory footprint https://dev.to/this-is-learning/performance-benchmarking-string-and-string-builder-3bid

Reply:
Fixed in 1490ed6.